src/MomentoTaggedCache.php

```diff
@@ -27,21 +34,32 @@ public function put($key, $value, $ttl = null)
     public function get($key, $default = null): mixed
     {
         $tags = $this->tags->getNames();
-        $cacheName = $this->store->getCacheName();
-        $value = null;
-        foreach ($tags as $tag) {
-            $keys = $this->store->setFetch($cacheName, $tag);
-            if (is_null($keys)) {
-                return $value;
-            } else {
-                foreach ($keys as $k) {
-                    if ($k == $key) {
-                        $value = $this->store->get($key);
-                        break;
-                    }
-                }
-            }
+        if (!self::validateTags($tags)) {
+            return null;
         }
-        return $value;
+        $newKey = self::createNewKey($tags, $key);
+        $hashedKey = hash("sha256", $newKey);
+        return $this->store->get($hashedKey);
```

Comment:
If the return value here is `null`, we're supposed to return `$default` instead. That variable is `null` by default but might be overridden by the user.

Reply:
Sorry forgot to address that. Thank you for pointing it out!

Reply:
I'll add an additional test to check this behavior as well.